frontend/src/Components/FavoriteCard.tsx

```diff
@@ -0,0 +1,71 @@
+import { CombinedFavorite } from '@/common';
+import { StarIcon as SolidStar } from '@heroicons/react/24/solid';
+import { useNavigate } from 'react-router-dom';
+
+interface FavoriteCardProps {
+    favorite: CombinedFavorite;
+    onUnfavorite: () => void;
+}
+
+const FavoriteCard: React.FC<FavoriteCardProps> = ({
+    favorite,
+    onUnfavorite
+}) => {
+    const navigate = useNavigate();
+
+    const handleCardClick = () => {
+        if (favorite.visibility_status) {
+            return;
+        }
+        const state = {
+            subLink: favorite.content_url
+        };
+        if (favorite.type === 'video') {
+            navigate(`/viewer/videos/${favorite.content_id}`);
+        } else if (favorite.type === 'library') {
+            navigate(`/viewer/libraries/${favorite.content_id}`, { state });
+        }
+    };
+
+    return (
+        <div
+            className={`card relative rounded-lg p-3 shadow-md ${
```

Comment:
`card` class should handle `rounded-lg shadow-md` so you dont have to repeat that in this class. you can see what is in the card class inside of `app.css`